### ISSUE TITLE
Add `+/-` operator support between time and duration

### ISSUE THREAD

**Issue by OfTheDelmer** [influxdata/flux#4544]
Depends on:  https://github.com/influxdata/flux/issues/4542

After implementing `date` package support for time and duration addition and subtraction we can build in operator support for these. This removes the need for having to import a package.

Considerations possibly already done here or in #4542:
 -  [Multi-parameter_type_classes](https://wiki.haskell.org/Multi-parameter_type_class)
 - [Functional Dependencies](https://wiki.haskell.org/Functional_dependencies)

**Comment by OfTheDelmer** [comment]
Could be related so I'm linking here for the curious:

See also: https://github.com/influxdata/flux/issues/2028
